lib/ApiHelper.js

```diff
@@ -116,6 +116,8 @@ class ApiHelper {
             }
             credentials = data.credentials;
             this.cred.sessionId = credentials.sessionId;
+        }).catch(err => {
+            throw err;
```

Comment:
@tranjennie What issue does catching and re-throwing the err fix? We should add a unit test to cover this senario if possible.

Reply:
@beefo I've removed this code. After some testing, I see that this was not catching any errors

Reply:
@tranjennie there are a bunch more of the same catch and re-thow paterns added.
```
 .catch(err => {
                    throw err;
                });
```
If the one above didn't do anything, I assume the other don't do anything?
If we remove all of these, there's no change here as far as I can tell. What precisely are we fixing here?

Reply:
If there is a legitimate fix here, please add a unit test for it.